site/sfguides/src/machine_learning_with_snowpark_python/machine_learning_with_snowpark_python.md

```diff
@@ -0,0 +1,429 @@
+    session.sql("CREATE USER IF NOT EXISTS "+username+\
+                " PASSWORD = '"+initial_password+\
+                "' LOGIN_NAME = '"+username+\
+                "' DEFAULT_ROLE = '"+project_role+\
+                "' MUST_CHANGE_PASSWORD = TRUE")\
+            .collect()
+    session.sql("GRANT ROLE "+project_role+" TO USER "+username).collect()
+
+session.use_role('sysadmin')
+session.sql("GRANT CREATE DATABASE ON ACCOUNT TO ROLE "+project_role).collect()
+
+session.use_role(state_dict['connection_parameters']['role'])
+
+session.use_role('accountadmin')
+project_role='PUBLIC'
+
+for wh in state_dict['compute_parameters'].values():
+    session.sql("CREATE WAREHOUSE IF NOT EXISTS "+wh+\
+            " WITH WAREHOUSE_SIZE = '"+wh.split('_')[0]+\
+            "' WAREHOUSE_TYPE = 'STANDARD' AUTO_SUSPEND = 300 AUTO_RESUME = TRUE initially_suspended = true;")\
+        .collect()
+    session.sql("GRANT USAGE ON WAREHOUSE "+wh+" TO ROLE "+project_role).collect() 
+    
+session.use_role(state_dict['connection_parameters']['role'])
+
+session.use_role('accountadmin')
+session.sql("GRANT IMPORT SHARE ON ACCOUNT TO "+project_role).collect()
+session.use_role(state_dict['connection_parameters']['role'])
+```
+
+<!-- ------------------------ -->
+## Data Engineering
+Duration: 10
+
+**Persona**: Data Engineeer
+
+**What You'll Do**: 
+- Extract the Citbike data from S3 to a Snowflake stage
+- Load the raw data to Snowflake tables
+- Create tables with transformed data for use by Data Scientists
+- Export the relevant DE code as Python modules and function for use in MLOps orchestration later
+
+**Snowflake Features**:
+- Snowpark Python Client Dataframe API for ELT
+- Snowflake Ingest Capabilities & Performance (loading ~94M records in roughly 30s)
+
+In this section of the demo, we will utilize Snowpark's Python client-side Dataframe API to build an ELT pipeline. We will extract the data from the source system (s3), load it into snowflake and add transformations to clean the data before analysis.
+
+Open up the [`01_Data_Engineering`](https://github.com/Snowflake-Labs/vhol-citibike-ml-snowpark-python/blob/main/01_Data_Engineering.ipynb) notebook and execute the cells. This notebook will:
+
+Now that we've done our initial engineering and loaded a sample of data, let's create some bulk ELT processes using the exported modules from the Data Engineering notebook.
+
+Open up the [`01_Ingest`](https://github.com/Snowflake-Labs/vhol-citibike-ml-snowpark-python/blob/main/01_Ingest.ipynb) notebook and walk through the cells to see how to incrementally and bulk load data from S3 into Snowflake tables.
+
+Run the last few cells of the notebook to bulk-load ~94M records into your Snowflake table in only about 30 seconds.
+
+<!-- ------------------------ -->
+## Data Marketplace
+Duration: 5
+
+**Persona**: Data Engineer
+
+**What You'll Do**: Use the Snowflake Data Marketplace to subscribe to weather data from Weather Source.
```

Comment:
@sfc-gh-cbaechtold Pl change Snowflake Data Marketplace to Snowflake Marketplace.